script/template/index.d.ts

```diff
@@ -0,0 +1,22 @@
+declare namespace load {
+  interface Dictionary {
+    /**
+     * A buffer for the affix file at `index.aff` in UTF-8.
+     */
+    aff: Buffer
+
+    /**
+     * A buffer for the dictionary file at `index.dic` in UTF-8.
+     */
+    dic: Buffer
+  }
+
+  type Callback = (error: NodeJS.ErrnoException | null, result: Dictionary) => void;
+}
+
+/**
+ * Load the dictionary.
+ */
+declare const load: (callback: load.Callback) => void
+
+export = load
```

Comment:
- Why are namespaces used? I don’t see a reason (here)?
- `error` is `undefined` in the code I believe?
- Is a callback type needed, `Dictionary` fine?
- Perhaps expand on the docs some more:
  ```
  aff: Buffer in UTF-8 for the affix file (defines the language, keyboard, flags, and more).
  dic: Buffer in UTF-8 for the dictionary file (contains words and flags applying to those words).
  dictionary: Object representing a hunspell dictionary, with `aff` and `dic` fields
  callback: Callback called when dictionary is loaded
  ```

Reply:
> * Why are namespaces used? I don’t see a reason (here)?

The namespace is used to export the `Dictionary` and `Callback` types.

> * `error` is `undefined` in the code I believe?

It wasn’t at the time of writing. This was changed in 5f2b26a81d108061d565ba49a414182b08e473e4

> * Is a callback type needed, `Dictionary` fine?

I think it looks a bit nicer than defining the callback inline in the `load` type definition, but I can remove it if you want.

> * Perhaps expand on the docs some more:
>   ```
>   aff: Buffer in UTF-8 for the affix file (defines the language, keyboard, flags, and more).
>   dic: Buffer in UTF-8 for the dictionary file (contains words and flags applying to those words).
>   dictionary: Object representing a hunspell dictionary, with `aff` and `dic` fields
>   callback: Callback called when dictionary is loaded
>   ```

Done. :+1: